cfpb/cfgov-refresh
# Hold: Add Ask block for portal pages

Adds a series of blocks that can be used to pull content directly from Ask answers when building out main sections (The basics, Key terms, Common issues, Know your rights, How to guides) on Consumer Tools portal pages.

## Additions

- Ask block on sublanding page
- Components of Ask block, including AskLinkItem, AskTextItem, LinkItem, and TextItem blocks
- Customized version of HeadingLevelBlock, as AskHeadingLevelBlock
- Simplified version of HeadingBlock, as AskHeadingBlock
- Templates for Ask block, LinkItem, and TextItem

## Testing

1. Check out branch and run migration
2. Add a new sublanding page
3. Add an `Ask block` to new sublanding page's Content section
4. Test using the three Ask block column types to replicate sections of the [Auto Loans portal page](https://www.consumerfinance.gov/consumer-tools/auto-loans/)
5. Use the `Ask text column` to replicate the first 2 items in `The Basics` section: 
    - The `text item` option can be used for the first link in the section, which is to a non-Ask page.
    - The `ask item` option can be used for the second link, which is to an Ask question. If a valid id for an Ask question is entered in the `answer id` field, the heading value will automatically be pulled from the Ask question's statement field (this field is probably currently blank). The heading can be overridden by entering content in the optional `heading` field.
6. Use the `Ask link column` to replicate the `Common issues`section:
    - Use the `Ask item` option to replicate the first link in this section. If a valid Ask answer id is entered, the `statement` field on the Ask answer will supply the `link text` value. It can be overridden by entering a value in the optional `link text` field
    - Add a new test link using the `link item` option
7. Use the `Info unit column` to replicate the `Submit a complaint` section


## Screenshots

### Ask block on sublanding page
<img width="1224" alt="screen shot 2018-10-15 at 12 44 28 pm" src="https://user-images.githubusercontent.com/778171/46974511-0cd97c00-d079-11e8-9cf5-219f21567ac7.png">

### Ask block
<img width="1252" alt="screen shot 2018-10-15 at 12 44 40 pm" src="https://user-images.githubusercontent.com/778171/46974524-19f66b00-d079-11e8-8666-2f6d2abc6ea5.png">

### Ask link column

<img width="1197" alt="screen shot 2018-10-15 at 12 45 31 pm" src="https://user-images.githubusercontent.com/778171/46974560-32ff1c00-d079-11e8-8c09-90821ae75e95.png">

#### Ask item and link item in Ask link column
<img width="1221" alt="screen shot 2018-10-15 at 12 45 15 pm" src="https://user-images.githubusercontent.com/778171/46974543-27abf080-d079-11e8-886f-3cfc0d832762.png">

### Ask text column
<img width="1184" alt="screen shot 2018-10-15 at 12 45 44 pm" src="https://user-images.githubusercontent.com/778171/46974591-53c77180-d079-11e8-8f81-28c930b5f8a1.png">

#### Ask item in Ask text column
<img width="853" alt="screen shot 2018-10-15 at 12 46 06 pm" src="https://user-images.githubusercontent.com/778171/46974603-5aee7f80-d079-11e8-9f16-b44fe9de162e.png">

#### Text item in Ask text column
<img width="808" alt="screen shot 2018-10-15 at 12 46 17 pm" src="https://user-images.githubusercontent.com/778171/46974611-5e820680-d079-11e8-8ae5-dac51714f9d8.png">


### Info unit column
<img width="1037" alt="screen shot 2018-10-15 at 12 46 47 pm" src="https://user-images.githubusercontent.com/778171/46974616-62ae2400-d079-11e8-87b0-bfd3a3b05a61.png">




## Notes
- The original request was for a simplified page type that could be used to replicate the existing Consumer Tools portal pages by users without extensive Wagtail experience & for a way to pull content directly from fields on Ask answers into sections (The basics, Common issues, etc) on the portal pages. In order to maintain the flexibility of the existing portal pages, this solution adds a simplified Ask block to the sublanding page type that can be used to build the relevant page sections and that pulls content automatically from the statement field on Ask answers.
- Some of the features in the Ask block are intended to support the [Student Loans portal page](https://www.consumerfinance.gov/consumer-tools/student-loans/), which repeats each of the Ask-focused sections (The basics, Common issues, etc) once for `Getting student loans` and again for `Repaying student loans`. 
    - The `intro` field makes a second block-level heading possible in an Ask block
    - The `anchor link` field enables linking to an Ask block
    - The `Text item` block's `Heading level` options include `bold` for situations where there are already H2, H3, and H4 section-level headings


## Todos

-

## Checklist


## Testing checklist

### Browsers


### Accessibility


### Other
